src/pymor/parameters/base.py

```diff
@@ -349,135 +350,98 @@ def __new__(cls, *args, **kwargs):
                 if vv.ndim == 0:
                     vv.shape = (1,)
                 assert vv.ndim == 1
                 assert k != 't' or len(vv) == 1
-            assert not vv.setflags(write=False)
-            values_for_t[k] = vv
+                assert not vv.setflags(write=False)
+                values[k] = vv
 
-        mu = super().__new__(cls, values_for_t)
-        mu._raw_values = raw_values
-        return mu
+        assert 't' not in values or not time_dependent_values, 'cannot specify "t" and have time-dependent values'
 
-    def is_time_dependent(self, param):
-        """Check whether the values for a given parameter depend on time.
+        self._values = values
+        self.time_dependent_values = FrozenDict(time_dependent_values)
 
-        This is the case when the value for `self[param]` was given by a |Function|
-        instead of a constant array.
-        """
-        from pymor.analyticalproblems.functions import Function
-        return isinstance(self._raw_values[param], Function)
+    def __getitem__(self, key):
+        return self._values[key]
 
-    def get_time_dependent_value(self, param):
-        """Return time-dependent |Function| for given parameter.
+    def __iter__(self):
+        return iter(self._values)
 
-        Parameters
-        ----------
-        param
-            The parameter for which to return the time-dependent values.
+    def __len__(self):
+        return len(self._values)
 
-        Returns
-        -------
-        If `param` depends on time, this corresponding |Function| (and not its
-        evaluation at the current time) is returned. If `param` is not given
-        by a |Function|, a |ConstantFunction| is returned.
-        """
-        from pymor.analyticalproblems.functions import Function
-        value = self._raw_values[param]
-        if not isinstance(value, Function):
-            from pymor.analyticalproblems.functions import ConstantFunction
-            value = ConstantFunction(value)
-        return value
+    def __contains__(self, key):
+        return key in self._values
+
+    def keys(self):
+        return self._values.keys()
+
+    def values(self):
+        return self._values.values()
 
-    def with_(self, **kwargs):
-        return Mu(self._raw_values, **kwargs)
+    def items(self):
+        return self._values.items()
+
+    def get(self, key, value=None):
+        return self._values.get(key, value)
 
     @property
-    def parameters(self):
-        return Parameters({k: v.size for k, v in self.items()})
+    def has_time_dependent_values(self):
+        return len(self.time_dependent_values) > 0
 
-    def allclose(self, mu):
-        """Compare dicts of |parameter values| using :meth:`~pymor.tools.floatcmp.float_cmp_all`.
+    def parameters(self, include_time_dependent=False):
+        params = {k: v.size for k, v in self.items()}
+        if include_time_dependent:
+            params.update({k: v.shape_range[0] for k, v in self.items()})
```

Comment:
```suggestion
            params.update({k: v.shape_range[0] for k, v in self.time_dependent_values})
```
Or do I misunderstand the idea of this method and in particular this line?

Reply:
You're right, but it should be 

```suggestion
            params.update({k: v.shape_range[0] for k, v in self.time_dependent_values.items()})
```